### PR TITLE
test assert & abort

### PR DESCRIPTION
**Pull request by Hecate2**
I am making use of the optimizer and assembly analyzers...
Will push more tests about abort & assert later.

### CODE REVIEW

**Comment by Jim8y** [comment]
This is not done yet by us, but may you please add comments to the test contract, explaining the expected behavior?

**Comment by Hecate2** [comment]
> This is not done yet by us, but may you please add comments to the test contract, explaining the expected behavior?

OK

File: src/Neo.Compiler.CSharp/Optimizer/DumpNef.cs [hunk 1]
```diff
@@ -174,6 +174,32 @@ string TryComment(int catchOffset, int finallyOffset)
             }
         }
 
+        public static (int start, int end) GetMethodStartEndAddress(string name, JToken debugInfo)
+        {
+            name = name[0].ToString().ToUpper() + name.Substring(1);  // first letter uppercase
```
Comment (shargon) [inline src/Neo.Compiler.CSharp/Optimizer/DumpNef.cs:179]:
```suggestion
            name = name.Length == 0 ? string.Empty : name[0].ToString().ToUpper() + name.Substring(1);  // first letter uppercase
```

**Review by shargon — APPROVED** [review]

**Comment by Jim8y** [comment]
@shargon also a suggestion to you, may you please add comments to every methods that you add to the code base? especially public methods. I am not going to require you of doing that, but suggest you to do it.

File: src/Neo.Compiler.CSharp/Optimizer/DumpNef.cs [hunk 2]
```diff
@@ -174,6 +174,32 @@ string TryComment(int catchOffset, int finallyOffset)
             }
         }
 
+        public static (int start, int end) GetMethodStartEndAddress(string name, JToken debugInfo)
+        {
+            name = name.Length == 0 ? string.Empty : name[0].ToString().ToUpper() + name.Substring(1);  // first letter uppercase
```
Comment (Jim8y) [inline src/Neo.Compiler.CSharp/Optimizer/DumpNef.cs:179]: @shargon Nullorempty?

**Pull request merged**